plugins/pencilblue/controllers/actions/admin/sites/new_site.js

```diff
@@ -48,7 +48,7 @@ module.exports = function NewSiteActionModule(pb) {
             var selectedLocale = self.body.selectedLocales[i];
             self.body.supportedLocales[selectedLocale] = true;
         }
-        self.body.selectedLocales = undefined;
+        delete self.body.selectedLocales;
```

Comment:
My suggestion is no.  0.5.0 has not been officially released so the majority of users wouldn't benefit from it.  You guys will be better off running a one time update against your environments' DB to do the cleanup.  Think of it as a schema change.